### PR TITLE
SPARK-4159 [BUILD] Addendum: improve running of single test after enabling Java tests

### PR DESCRIPTION
**Pull request by srowen**
https://issues.apache.org/jira/browse/SPARK-4159 was resolved but as Sandy points out, the guidance in https://cwiki.apache.org/confluence/display/SPARK/Useful+Developer+Tools under "Running Individual Tests" no longer quite works, not optimally.

This minor change is not really the important change, which is an update to the wiki text. The correct way to run one Scala test suite in Maven is now:

```
mvn test -DwildcardSuites=org.apache.spark.io.CompressionCodecSuite -Dtests=none
```

The correct way to run one Java test is

```
mvn test -DwildcardSuites=none -Dtests=org.apache.spark.streaming.JavaAPISuite
```

Basically, you have to set two properties in order to suppress all of one type of test (with a non-existent test name like 'none') and all but one test of the other type.

The change in the PR just prevents Surefire from barfing when it finds no "none" test.
I'd make the wiki edit but I can't myself.

### CODE REVIEW

**Comment by JoshRosen** [comment]
@pwendell @mateiz Could one of you grant @srowen edit permissions on the Spark CWiki?  I'd do it myself but I don't think I have admin privileges (could you grant them to me?).

**Comment by sryza** [comment]
@srowen do you know whether the sbt test instructions are still correct?

**Comment by sryza** [comment]
made the wiki edit for Maven

**Comment by mateiz** [comment]
Gave Sean edit permissions and made you admin. FYI the admin panel is really hard to find, it's under Browse -> Space Admin.

**Comment by srowen** [comment]
@sryza SBT won't run the Java tests, so that is unchanged. Wiki edit looks good to me. We do need this PR merged too or else excluding all Java tests will cause Surefire to barf.

**Comment by JoshRosen** [comment]
I think that SBT actually _can_ run the Java tests, since I've successfully run only `JavaAPISuite` through SBT.

This looks good to me, so I'm going to merge this into `master` (1.3.0).  Thanks!
